packages/core/src/textures/TextureSystem.js

```diff
@@ -68,6 +49,17 @@ export default class TextureSystem extends System
 
         this.CONTEXT_UID = this.renderer.CONTEXT_UID;
 
+        let i;
+
+        const maxTextures = gl.getParameter(gl.MAX_TEXTURE_IMAGE_UNITS);
+
+        this.boundTextures.length = 0;
```

Comment:
You should set this to `boundTextures` so the engine doesn't create a sparse array later when you assign to out-of-bounds indices.